prefetch.bs

```diff
@@ -252,233 +325,283 @@ A <dfn export>cross-origin prefetch IP anonymization policy</dfn> has an <dfn ex
+        :  [=navigation params/response=]
+        :: |response|
+        :  [=navigation params/origin=]
+        :: |responseOrigin|
+        :  [=navigation params/policy container=]
+        :: |resultPolicyContainer|
+        :  [=navigation params/final sandboxing flag set=]
+        :: |finalSandboxFlags|
+        :  [=navigation params/cross-origin opener policy=]
+        :: |responseCOOP|
+        :  [=navigation params/COOP enforcement result=]
+        :: |coopEnforcementResult|
+        :  [=navigation params/reserved environment=]
+        :: |request|'s [=request/reserved client=]
+        :  [=navigation params/navigable=]
+        :: |navigable|
+        :  [=navigation params/navigation timing type=]
+        :: |navTimingType|
+        :  [=navigation params/fetch controller=]
+        :: |fetchController|
+        :  [=navigation params/commit early hints=]
+        :: |commitEarlyHints|
 </div>
 
+<h2 id="prefetch-algorithms">Prefetch algorithms</h2>
+
+<p class="issue">Check Service Worker integration</p>
+
 The <dfn>list of sufficiently strict speculative navigation referrer policies</dfn> is a list containing the following: "", "`strict-origin-when-cross-origin`", "`strict-origin`", "`same-origin`", "`no-referrer`".
 
-<div algorithm="uncredentialed prefetch">
-    To <dfn export>uncredentialed prefetch</dfn> given a {{Document}} |document| and [=prefetch record=] |prefetchRecord|, perform the following steps.
+<div algorithm>
+    To <dfn export>prefetch</dfn> given a {{Document}} document and a [=prefetch record=] |prefetchRecord|, perform the following steps.
+
+    1. Let |sourcePartitionKey| be the result of [=determining the network partition key=] given |document|'s [=relevant settings object=].
+    1. Let |sourceSnapshotParams| be the result of [=snapshotting source snapshot params=] given |document|.
+    1. Let |targetSnapshotParams| be the result of [=snapshotting target snapshot params=] given |document|'s [=node navigable=].
+    1. Let |topLevelOrigin| be |prefetchRecord|'s [=prefetch record/URL=]'s [=url/origin=] if |document|'s [=node navigable=] is a [=top-level traversable=], and |document|'s [=relevant settings object=]'s [=environment/top-level origin=] otherwise.
+    1. Let |topLevelSite| be the result of [=obtaining a site=] given |topLevelOrigin|.
+    1. Let |secondKey| be null or an [=implementation-defined=] value.
+    1. Let |partitioningScheme| be "`same-partition`" if |sourcePartitionKey| is equal to (|topLevelSite|, |secondKey|), and "`uncredentialed`" otherwise.
+
+        <div class="note">
+            This determines whether the navigation would use the same network partition key. If it would, the prefetch is restricted to the same partition, and redirects which would leave the partition cause the prefetch to fail.
 
+            Otherwise it is uncredentialed, and redirects which would return to the original partition (thus ought to have credentials) will cause prefetch to fail.
```

Comment:
```suggestion
            <p>Otherwise it is uncredentialed, and redirects which would return to the original partition (thus ought to have credentials) will cause prefetch to fail.
```